tools/val/val_strings.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TOOLS_VAL_VAL_STRINGS_H
+#define TOOLS_VAL_VAL_STRINGS_H
+
+namespace spvtools {
+namespace val {
+
+const char* shared_validator_options_string = R"(
+  --relax-logical-pointer          Allow allocating an object of a pointer type and returning
+                                   a pointer value from a function in logical addressing mode
+  --relax-block-layout             Enable VK_KHR_relaxed_block_layout when checking standard
+                                   uniform, storage buffer, and push constant layouts.
+                                   This is the default when targeting Vulkan 1.1 or later.
+  --scalar-block-layout            Enable VK_EXT_scalar_block_layout when checking standard
+                                   uniform, storage buffer, and push constant layouts.  Scalar layout
+                                   rules are more permissive than relaxed block layout so in effect
+                                   this will override the --relax-block-layout option.
+  --skip-block-layout              Skip checking standard uniform/storage buffer layout.
+                                   Overrides any --relax-block-layout or --scalar-block-layout option.
+  --relax-struct-store             Allow store from one struct type to a
+                                   different type with compatible layout and
+                                   members.)";
+
+}  // namespace val
```

Comment:
My concern about this is that it's not future-proof.  These options are shared with spirv-reduce, but they could end up getting shared with other tools.  And then maybe not all of them would be shared - perhaps a subset of them, or a partially-overlapping set that isn't a subset.  Also the name "shared_validator_options_string" doesn't indicate what they are shared with, yet it would be bad to have the validator refer explicitly to the reducer.

I suggest instead having one const char* per option:

const char* validator_option_relax_logical_pointer = ...;

const char* validator_option_relax_block_layout = ...;

etc.

This would make it possible to expose combinations of option help strings to other tools in a fine-grained manner and would avoid the problem of documenting vs. not documenting in the validator the fact that these options are shared.

Side note: are there naming conventions for string constants, such as starting their names with "k"?  Or is that just for numeric constants?

Reply:
@alan-baker I'd be curious to know what you think about this?  Options are:

- have relevant validator options text duplicated between validator and reducer
- factor out the text like Paul has done here
- avoid having the reducer explain those validator options at all, and instead have it say: "See spirv-val --help for details on the following validator options:"

Reply:
I think for now I'd prefer something like the third option:
```
--relax-block-layout . Validate using Vulkan relaxed block layout. See spirv-val --help for a detailed description
```

Then we can consider how we should best share options separately from this PR.

Reply:
I have done the following to avoid repeating the help text. See what you think. 

```
Supported validator options are as follows. See `spirv-val --help` for details.
  --relax-logical-pointer
  --relax-block-layout
  --scalar-block-layout
  --skip-block-layout
  --relax-struct-store
```